src/App.tsx

```diff
@@ -15,35 +18,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  default = '',
+  length = 'length',
+  alphabetically = 'alphabetically',
+  reverse = 'reverse',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  reverseField: string,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverseField === SortType.reverse
+    ? preparedGoods.reverse()
+    : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [sortFieldReverse, setSortFieldReverse] = useState<string>('');
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
it can be a boolean state